sui-execution/v1/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1408,12 +1409,12 @@ mod checked {
                 let info_opt = primitive_serialization_layout(context, inner)?;
                 info_opt.map(|layout| PrimitiveArgumentLayout::Vector(Box::new(layout)))
             }
-            Type::StructInstantiation(struct_inst) => {
-                let (idx, targs) = &**struct_inst;
-                let Some(s) = context.vm.get_runtime().get_struct_type(*idx) else {
+            Type::DatatypeInstantiation(inst) => {
+                let (idx, targs) = &**inst;
+                let Some(s) = context.vm.get_runtime().get_type(*idx) else {
                     invariant_violation!("Loaded struct not found")
```

Comment:
Is this error wrong now?

Reply:
Both of these errors are actually still correct -- they are previous execution versions, so only run historic code and will never interact with enums (and only structs).